### PR TITLE
Add tflite-micro and some other AI support

### PR DESCRIPTION
**Pull request by jihandong**
## Summary
Recently, we ported tflite-micro on nuttx-apps with an API demo, it's a C++11 library for AI inference.

## Impact
Totally new feature, no one use it yet.

## Testing
CONFIG_TFLITEMICRO
CONFIG_TFLITEMICRO_TOOL

### CODE REVIEW

**Comment by cederom** [comment]
macOS build error seems CI problem also occurring on other builds unrelated to this change.

**Comment by jihandong** [comment]
> Thank you @jihandong amazing work!! :-)
> 
> Would it be possible to add Documentation and example / test applications in a separate PR in a free moment? :-)

I'd like to. But compared to the background knowledge of AI, tflite-micro is very esay to use and there are already many good documents. Share some steps here:
1. First u need to prepare your own tflite model file.
2. Register needed operators by `tflite::MicroMutableOpResolver<T>::Add*`.
3. Prepare a piece of memory as memory pool(tflite-micro calls it "arena") for intermediate calculations.
4. Initialize a `MicroIntepreter`, every instance represents a parsed model.
5. Finally, ou can get the input and output addresses(from the memory pool) through `interpreter::GetInput` and `interpreter::GetOutput`, `interpreter::Invoke` would do once inference.

Hope these steps help u. :)

**Comment by cederom** [comment]
> > Thank you @jihandong amazing work!! :-)
> > Would it be possible to add Documentation and example / test applications in a separate PR in a free moment? :-)
> 
> I'd like to. But compared to the background knowledge of AI, tflite-micro is very esay to use and there are already many good documents. Share some steps here:
> 
>     1. First u need to prepare your own tflite model file.
> 
>     2. Register needed operators by `tflite::MicroMutableOpResolver<T>::Add*`.
> 
>     3. Prepare a piece of memory as memory pool(tflite-micro calls it "arena") for intermediate calculations.
> 
>     4. Initialize a `MicroIntepreter`, every instance represents a parsed model.
> 
>     5. Finally, ou can get the input and output addresses(from the memory pool) through `interpreter::GetInput` and `interpreter::GetOutput`, `interpreter::Invoke` would do once inference.
> 
> 
> Hope these steps help u. :)

Its not really for me, but for any user or newcomer to NuttX that would be interested in this subject :-) 

We have project documentation available at:

https://nuttx.apache.org/docs/latest/

Source code for this Documentation is here:

https://github.com/apache/nuttx/tree/master/Documentation

We usually provide starting point "examples" that presents some basic functionalities and acts as project templates:

https://github.com/apache/nuttx-apps/tree/master/examples

Its also good way to announce your work results to the world :-)

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by anchao** [comment]
let us remove VELAPLATFORM and change-id from commit message:
image

File: mlearning/tflite-micro/operators/neon/arm_nn_mat_mult_kernel_s8_s16.c [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+#include "arm_nnfunctions.h"
+#include "arm_nnsupportfunctions.h"
+
+/*
+ * Matrix-multiplication function for convolution with per-channel requantization.
+ *
+ * Refer header file for details.
+ *
+ */
+int8_t *arm_nn_mat_mult_kernel_s8_s16(const int8_t *input_a,
```
Comment (anchao) [inline mlearning/tflite-micro/operators/neon/arm_nn_mat_mult_kernel_s8_s16.c:49]: why not call cmsis-nn or ne10 directly?

Comment (xiaoxiang781216) [reply mlearning/tflite-micro/operators/neon/arm_nn_mat_mult_kernel_s8_s16.c:49]: cmsis-nn doesn't support neon, nobody upstream ne10 library.

**Pull request merged**